File: native/apple/Sources/IosAwnCore/AwesomeEventListener.swift
import Foundation

/// Listener for notification lifecycle events. Mirrors the original IosAwnCore
/// `AwesomeEventListener`: the Flutter plugin implements it and forwards each
/// event to the method channel.
public protocol AwesomeEventListener: AnyObject {
    func onNewAwesomeEvent(eventType: String, content: [String: Any])
}

File: native/android/AndroidAwnCore/src/main/kotlin/me/carda/android_awn_core/AwesomeEventListener.kt
package me.carda.android_awn_core

/**
 * Listener for notification lifecycle events. Mirrors the original
 * AndroidAwnCore `AwesomeEventListener`: the Flutter plugin implements it and
 * forwards each event to the method channel.
 */
interface AwesomeEventListener {
    fun onNewAwesomeEvent(eventType: String, content: Map<String, Any?>)
}
